sdcm/nemesis.py

```diff
@@ -1725,13 +1727,44 @@ def disrupt_nodetool_cleanup(self):
             for keyspace in test_keyspaces:
                 node.run_nodetool(sub_cmd="cleanup", args=keyspace)
 
-    def _prepare_test_table(self, ks='keyspace1', table=None):
+    def _prepare_test_table(self,
+                            ks: str = 'keyspace1',
+                            table: str | None = None,
+                            pre_create_objects: bool = False) -> None:
         ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node)
         table_exist = f'{ks}.{table}' in ks_cfs if table else True
```

Comment:
we can drop the following lines:
```
ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node)
        table_exist = f'{ks}.{table}' in ks_cfs if table else True
```

it because both create keyspace and create table they use `IF NOT EXISTS`, so in this case we can get drop querying, because it may slow the whole prepare/create part

Reply:
Yes, I understand what `IF NOT EXISTS` does, but the change you suggest affects the logic of `_prepare_test_table`.
Now we do not start the C-S thread if both the keyspace and the table exist.
If I remove these lines the C-S will be triggered each time, regardless of the objects' presence. If the objects exist it will at least try to write data to them each time.
I am not sure how It affects all the nemeses using  `_prepare_test_table` - this is not only `NoCorruptRepairMonkey`, but also `SwitchBetweenPasswordAuthAndSaslauthdAuth`, `LoadAndStreamMonkey`, `RefreshBigMonkey`, `RefreshMonkey`, `TruncateMonkey`.
What do you think?

Reply:
if you create the keyspace and table using `IF NOT EXISTS`, then there is no need to query if they exist before, and if you populate the tables using CQL command through Python, then you also don't need to use `c-s` for it... 
all these nemeses will most likely want to have content in their tables, but not sure they need something specific in the keyspace or table...

Reply:
@fgelcer just to be on the same page with you.
Let's say we already have the keyspace and table before calling `_prepare_test_table`
In this case, the code
```python
        ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node)
        table_exist = f'{ks}.{table}' in ks_cfs if table else True

        test_keyspaces = self.cluster.get_test_keyspaces()
        # If keyspace or table doesn't exist, create it
        if ks not in test_keyspaces or not table_exist:
            ...
```
prevents populating this table with data, right?
If I remove this code (because now I am relying on `IF NOT EXISTS` and don't need extra querying), then I will populate the table no matter whether it existed before (empty or not) or has just been created.
It looks like a change of `_prepare_test_table` algorithm that may affect other nemeses, doesn't it?

Sorry for asking dumb questions, but now I am just trying to understand which way to proceed.
From my understanding, initially, I was supposed to create a simple workaround for `disrupt_no_corrupt_repair` nemesis.
But now it looks like I am about to significantly change `_prepare_test_table`. After this change `_prepare_test_table` will behave a bit differently, which means this workaround impacts all the nemeses using it. In my opinion, this impact needs to be tested.
I can do that, no problem.
I just want to clarify the goal I need to reach working on this fix.

Reply:
As agreed, the only thing we need in this workaround is to wait for the schema agreement in `disrupt_no_corrupt_repair`.
The proposed change is not needed.